src/vsr/replica.zig

```diff
@@ -4637,6 +4672,23 @@ pub fn ReplicaType(
             assert(self.op == message.header.op);
         }
 
+        fn primary_prepare_reconfiguration(self: *const Self, request: *Message) void {
+            assert(self.primary());
+            assert(request.header.command == .request);
+            assert(request.header.operation == .reconfigure);
+            assert(request.header.size == @sizeOf(vsr.Header) + @sizeOf(vsr.ReconfigurationRequest));
+            var reconfiguration_request = std.mem.bytesAsValue(
+                vsr.ReconfigurationRequest,
+                request.body()[0..@sizeOf(vsr.ReconfigurationRequest)],
+            );
+            reconfiguration_request.*.result = reconfiguration_request.validate(.{
```

Comment:
Here, we curiously take advantage of our ability to do some computation during prepare, rather than commit. 

For reconfiguration, primary needs to make a decision whether its good or not during prepare, not during commit; preparing reconfigration takes effect immediately on the primary, as it must stop processing any further requests.